spec.html

```diff
@@ -7435,9 +7435,9 @@ <h1>[[Construct]] ( _argumentsList_, _newTarget_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-functionallocate" aoid="FunctionAllocate">
-      <h1>FunctionAllocate ( _functionPrototype_, _functionKind_ )</h1>
-      <p>The abstract operation FunctionAllocate requires the two arguments _functionPrototype_ and _functionKind_. FunctionAllocate performs the following steps:</p>
+    <emu-clause id="sec-ordinaryfunctioncreate" aoid="OrdinaryFunctionCreate" oldids="sec-functionallocate sec-functioninitialize">
+      <h1>OrdinaryFunctionCreate ( _functionPrototype_, _ParameterList_, _Body_, _functionKind_, _kind_, _Scope_ )</h1>
+      <p>The abstract operation OrdinaryFunctionCreate requires the arguments: an object _functionPrototype_, a parameter list Parse Node specified by _ParameterList_, a body Parse Node specified by _Body_, _functionKind_ which is one of (`"normal"`, `"non-constructor"`, `"generator"`, `"async"`, or `"async generator"`), _kind_ which is one of (~Normal~, ~Method~, ~Arrow~), and a Lexical Environment specified by _Scope_. OrdinaryFunctionCreate performs the following steps:</p>
```

Comment:
what is "non-constructor" - is that just "arrow or concise"?

Reply:
OrdinaryFunctionCreate's `_functionKind_` parameter has exactly the same semantics as FunctionAllocate's `_functionKind_` parameter in the current spec, so "non-constructor" means the same in both.

The only place that passes "non-constructor" is FunctionCreate, and only if its `_kind_` arg isn't `~Normal~`, so an ArrowFunction, or MethodDefinition (except for a class constructor?).

And the effect of "non-constructor" appears to be the same as "normal" except that the function's `[[Construct]]` and `[[ConstructorKind]]` aren't set.

Note that PR #1546 replaces the `_functionKind_` parameter with a direct `_needsConstruct_` parameter, which causes `"non-construct"` and its brethren to disappear.